dafni_cli/commands/get.py

```diff
@@ -165,16 +212,23 @@ def dataset(ctx: Context, id: str, version_id: str, long: bool, version_history:
         version_id (str): Dataset version ID
         long (bool): Flag to view additional metadata attributes
         version_history (bool): Flag to view version history in place of metadata
+        json (bool): Flag to view json returned from API
     """
     metadata = get_latest_dataset_metadata(ctx.obj["jwt"], id, version_id)
     if not version_history:
-        dataset_meta = dataset_metadata.DatasetMetadata(metadata)
-        dataset_meta.output_metadata_details(long)
+        if json:
+            print_json(metadata)
+        else:
+            dataset_meta = dataset_metadata.DatasetMetadata(metadata)
+            dataset_meta.output_metadata_details(long)
     else:
-        version_history = dataset_version_history.DatasetVersionHistory(
-            ctx.obj["jwt"], metadata
-        )
-        version_history.process_version_history()
+        if json:
```

Comment:
I think for this one, this may be the wrong place to do this. Probably better to add it as an arg into ` process_version_history` so that the dataset details are displayed for each version, instead of a a dict with just version ids.
so:
`
        for version_id in self.version_ids:
            metadata = get_latest_dataset_metadata(
                self.jwt, self.dataset_id, version_id
            )
            dataset_meta = DatasetMetadata(metadata)
            dataset_meta.output_version_details()
`
to:
`        for version_id in self.version_ids:
            metadata = get_latest_dataset_metadata(
                self.jwt, self.dataset_id, version_id
            )
            if json:
                print_json(metadata)
            else:
                dataset_meta = DatasetMetadata(metadata)
                dataset_meta.output_version_details()
`

Reply:
This has been changed, with a slight variation in that the metadata json outputs are printed as a list, as opposed to one after the other.